src/util/transaction_identifier.h

```diff
@@ -36,7 +36,10 @@ class transaction_identifier
     }
 
 public:
+    using value_type = uint256::value_type;
+
     transaction_identifier() : m_wrapped{} {}
+    transaction_identifier(std::span<const value_type> sp) : m_wrapped(sp) {}
```

Comment:
Is there any specific reason not to use `unsigned char` directly, like `std::span<const unsigned char > sp)` ?

Reply:
In theory we could change `uint256`'s `value_type`, and we wouldn't need to make any changes here in that case.

Reply:
I've dropped this commit from this PR and moved it to #33034. Also dropped `value_type` and made the constructor just take `span<const std::byte>` and that seems to work.